### ISSUE TITLE
No alarm on pump disconnection (Combo)

### ISSUE THREAD

**Issue by hartlijn** [MilosKozak/AndroidAPS#1603]
AndroidAPS v2.0 flavor: fullrelease build 2018.11.05
Combo pump
[AndroidAPS._2019-01-16_00-00-22_.0.zip](https://github.com/MilosKozak/AndroidAPS/files/2766225/AndroidAPS._2019-01-16_00-00-22_.0.zip)
once in a while I loose connection to the pump. this is a known issue. Normally I get an error message in AAPS as I have an alarm set (Alert if pump is unreachable is ON with 31 min treshold)

This night the connection got lost without notification, keeping my pump running without AAPS connection (pushing my BG down, until I noticed when I woke up).
No Careportal message in the app can be found (as I do find normally when connection got lost).
Logs attached. It must have happened around 01:23h (16 jan), looking at timestamp of the phone and the message in Combo screen (326 minutes no connection).

![image](https://user-images.githubusercontent.com/28272471/51281085-b23a1e00-19e1-11e9-8b1d-939be4abb4db.png)
![image](https://user-images.githubusercontent.com/28272471/51281104-bf570d00-19e1-11e9-8cf7-641a9df10c23.png)

**Comment by MilosKozak** [comment]
@jotomo ?

**Comment by jotomo** [comment]
Looked at this multiple times ... combo bails due to comm issues, after that _KeepAliveReceiver_ stops requesting status updates, which should only only be possible if the pump claims to be busy perpetually. Which should really not be possible.
tl;dr; tricky to reproduce, if at all, and no idea how address this.
@hartlijn Has this ever happened again? And what phone and OS are you using?

**Comment by Chris2001** [comment]
Similar problem here with ver. 2.2.2 now, no alert even after 3 hrs no connection, local alert set to 31 mins.
With Dana RS, Sony XA1, Android 7
Local Alert for missed BG readings after 45 mins (e.g. during sesonr warm up) works just fine.

**Comment by Woobagcgm** [comment]
I have the same problem. 

Samsung Galaxy A6
Android 10
Accu Chek Combo
AAPS 2.3

No alarms when this happens, even after hours (also during the night, see second screenshot), reconnection only works when AAPS is closed or phone reboot and then open AAPS again.
Screenshot is from NS when it happens today again after lunch. New battery yesterday.
![Bild 23 09 19 um 15 40](https://user-images.githubusercontent.com/31259305/65430854-e1c6e300-de18-11e9-88cd-2d35d7fc86e0.jpeg)

![Bild 23 09 19 um 15 50](https://user-images.githubusercontent.com/31259305/65431743-3f0f6400-de1a-11e9-85b2-bc83059cf222.jpeg)
After this night, pump battery was 25%.

**Comment by hartlijn** [comment]
.

**Comment by kropas13** [comment]
Same problem since updating from 3.0 to 3.1.0.4. 
The Combo tab shows time since last connection (in red), but there is no alarm.
Pump unreachable threshold = 31 min
